p2p/host/eventbus/basic_metrics.go

```diff
@@ -0,0 +1,44 @@
+package eventbus
+
+import (
+	"reflect"
+	"sync"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const metricNamespace = "libp2p_eventbus"
+
+var (
+	eventsEmitted = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: metricNamespace,
+			Name:      "events_emitted_total",
+			Help:      "Events Emitted",
+		},
+		[]string{"event"},
+	)
+)
+
+type MetricsTracer interface {
+	EventEmitted(typ reflect.Type)
+}
+
+type metricsTracer struct{}
+
+var _ MetricsTracer = &metricsTracer{}
+
+var initMetricsOnce sync.Once
+
+func initMetrics() {
+	prometheus.MustRegister(eventsEmitted)
+}
+
+func NewMetricsTracer() MetricsTracer {
+	initMetricsOnce.Do(initMetrics)
+	return &metricsTracer{}
+}
+
+func (m *metricsTracer) EventEmitted(typ reflect.Type) {
+	eventsEmitted.WithLabelValues(typ.String()).Inc()
```

Comment:
Should we filter out the package name here (the `event.`)?

This is how it looks like in Grafana:
![image](https://user-images.githubusercontent.com/1478487/216177149-4d591567-1749-4508-b663-3f54402d7c28.png)

Reply:
fixed.